sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -1200,6 +1232,7 @@ private async Task RunAsync(CancellationToken cancellationToken)
                                                                    IEnumerable<PartitionOwnership> completeOwnershipEnumerable,
                                                                    CancellationToken cancellationToken)
         {
+            Logger.ClaimOwnership(partitionId);
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
```

Comment:
I would put it **after** the `ThrowIfCancellationRequested` call. If we were to cancel the request here, it would mean the ownership claim had never started in the first place.